lms/static/scripts/frontend_apps/components/ContentSelector.tsx

```diff
@@ -304,7 +291,7 @@ export default function ContentSelector({
           authToken={authToken}
           listFilesApi={moodleListPagesApi}
           onCancel={cancelDialog}
-          onSelectFile={selectMoodlePage}
+          onSelectFile={(file: File) => selectContent(file, 'Moodle page')}
```

Comment:
Typescript can infer the type of `file` here when defining the callback inline, because of the type defined for the `onSelectFile` prop.

```suggestion
          onSelectFile={file => selectContent(file, 'Moodle page')}
```

Reply:
On a second thought, it could be the `onSelectFile` prop defined the `file` param as `File | Folder`, in which case you would in fact need to narrow it down as you did here.